### ISSUE TITLE
TestMfa Failures in Vault 1.11

### ISSUE THREAD

**Issue by colin-pm** [hvac/hvac#859]
Vault 1.11 introduced the following errors in test_mfa.py. Creating an issue to track the resolution of these failures.

```
FAILED tests/integration_tests/api/auth_methods/test_mfa.py::TestMfa::test_configure_0_enable_mfa_with_supported_auth_method - hvac.exceptions.InvalidPath: 1 error occurred:
FAILED tests/integration_tests/api/auth_methods/test_mfa.py::TestMfa::test_configure_4_enable_mfa_with_unsupported_auth_method_type_forced - hvac.exceptions.InvalidPath: 1 error occurred:
FAILED tests/integration_tests/api/auth_methods/test_mfa.py::TestMfa::test_configure_duo_access_0_configure_duo_access_success - hvac.exceptions.InvalidPath: 1 error occurred:
FAILED tests/integration_tests/api/auth_methods/test_mfa.py::TestMfa::test_configure_duo_behavior_0_enable_mfa_with_supported_auth_method - hvac.exceptions.InvalidPath: 1 error occurred:
FAILED tests/integration_tests/api/auth_methods/test_mfa.py::TestMfa::test_login_with_mfa_0_login_without_duo_access_configured - hvac.exceptions.InvalidPath: 1 error occurred:
FAILED tests/integration_tests/api/auth_methods/test_mfa.py::TestMfa::test_read_configuration_0_read_configured_path - hvac.exceptions.InvalidPath: 1 error occurred:
FAILED tests/integration_tests/api/auth_methods/test_mfa.py::TestMfa::test_read_duo_behavior_configuration_0_read_configured_path - hvac.exceptions.InvalidPath: 1 error occurred:
```

**Comment by colin-pm** [comment]
Ah, I figured out the issue. The mfa authentication method that Vault interfaces with was marked as deprecated after the new mfa authentication mechanism was introduced in Vault 1.10. The legacy authentication method was removed in Vault 1.11.

https://www.vaultproject.io/docs/auth/login-mfa/faq#q-what-is-the-legacy-mfa-feature

Reworking hvac to use the new mfa mechanism seems like a lot of work and probably shouldn't be scoped into the HVAC 1.0.0 release.

I think the best option is to copy HVAC's mfa module to a legacy_mfa module and mark all the functionality in mfa as deprecated.  I think we can leave the legacy_mfa module in for those that want to use the newer version of HVAC with an older instance of Vault. We can then work on updating hvac to support the new mfa module for the next major release.

Curious about your thoughts @briantist and @adammike ?
